packages/valory/skills/registration_abci/behaviours.py

```diff
@@ -67,8 +122,277 @@ def async_act(self) -> Generator:
+        if not service_info:
+            return False
+
+        registered_addresses = set(service_info["agent_instances"])
+        if not registered_addresses:
+            log_message = self.LogMessages.no_agents_registered.value
+            self.context.logger.info(f"{log_message}: {service_info}")
+            return False
+
+        my_address = self.context.agent_address
+        if my_address not in registered_addresses:
+            log_message = f"{self.LogMessages.self_not_registered} ({my_address})"
+            self.context.logger.info(f"{log_message}: {registered_addresses}")
+            return False
+
+        # put service info in the shared state for p2p message handler
+        info: Dict[str, Dict[str, str]] = dict.fromkeys(registered_addresses)
+        validator_config = dict(
+            tendermint_url=self.context.params.tendermint_url,
+            address=self.local_tendermint_params["address"],
+            pub_key=self.local_tendermint_params["pub_key"],
+        )
+        info[self.context.agent_address] = validator_config
+        self.synchronized_data.db.update(registered_addresses=info)
+        log_message = self.LogMessages.response_service_info.value
+        self.context.logger.info(f"{log_message}: {info}")
+        return True
+
+    def get_tendermint_configuration(self) -> Generator[None, None, bool]:
+        """Make HTTP GET request to obtain agent's local Tendermint node parameters"""
+
+        url = self.tendermint_parameter_url
+        log_message = self.LogMessages.request_personal
+        self.context.logger.info(f"{log_message}: {url}")
+
+        result = yield from self.get_http_response(method="GET", url=url)
+        response = self._decode_result(result, self.LogMessages.failed_personal)
+        if response is None:
+            return False
+
+        self.local_tendermint_params = response["params"]
+        log_message = self.LogMessages.response_personal
+        self.context.logger.info(f"{log_message}: {response}")
+        return True
+
+    def request_tendermint_info(self) -> bool:
+        """Request Tendermint info from other agents"""
+
+        still_missing = {k for k, v in self.registered_addresses.items() if not v}
+        log_message = self.LogMessages.request_others
+        self.context.logger.info(f"{log_message}: {still_missing}")
+
+        for address in still_missing:
+            dialogues = cast(TendermintDialogues, self.context.tendermint_dialogues)
+            performative = TendermintMessage.Performative.REQUEST
+            message, _ = dialogues.create(
+                counterparty=address, performative=performative
+            )
+            message = cast(TendermintMessage, message)
+            context = EnvelopeContext(connection_id=P2P_LIBP2P_CLIENT_PUBLIC_ID)
+            self.context.outbox.put_message(message=message, context=context)
+
+        if all(self.registered_addresses.values()):
```

Comment:
A little delay before checking?

Reply:
@Karrenbelt I think this is a valid point and we might need to wait for the message after all?

Reply:
Reverted the change in e5bee6622139e56c34ff2f5928bfa5e0ca9a5b8b and also added a comment in 4f84c517a1de82049e4497c45296589bc6aaaba6.